salt/modules/yumpkg.py

```diff
@@ -580,7 +580,7 @@ def latest_version(*names, **kwargs):
     cmd = [_yum(), '--quiet']
     cmd.extend(repo_arg)
     cmd.extend(exclude_arg)
-    cmd.extend(['list', 'available'])
+    cmd.extend(['list'])
```

Comment:
According to the yum manpage:

```
       yum list [all | glob_exp1] [glob_exp2] [...]
              List all available and installed packages.

       yum list available [glob_exp1] [...]
              List  all  packages  in  the  yum  repositories  available  to   be
              installed.
```

Given that the [stated behavior of the pkg.latest_version function](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.yumpkg.html#salt.modules.yumpkg.latest_version) is not to return a version when the package is already installed, the code is already working as intended. This change would in fact _break_ the ``pkg.latest`` state for platforms using yum/dnf, and as such, I can't recommend merging this pull request until it is removed.